pkg/apis/core/validation/seed.go

```diff
@@ -270,6 +271,25 @@ func validateSeedNetworks(seedNetworks core.SeedNetworks, fldPath *field.Path, i
 	if seedNetworks.Nodes != nil {
 		networks = append(networks, cidrvalidation.NewCIDR(*seedNetworks.Nodes, fldPath.Child("nodes")))
 	}
+
+	if seedNetworks.VPN != nil {
+		// vpn network must be a valid network and must match the primary IP family (see below)
+		vpnNetwork = cidrvalidation.NewCIDR(*seedNetworks.VPN, fldPath.Child("vpn"))
+		// vpn network must be disjoint with the other networks (see below)
+		networks = append(networks, vpnNetwork)
+
+		// vpn network must be a /24 network for IPv4 and /120 for IPv6
+		if vpnCIDR := vpnNetwork.GetIPNet(); vpnCIDR != nil {
+			prefix, _ := vpnCIDR.Mask.Size()
+			if primaryIPFamily == core.IPFamilyIPv4 && prefix != 24 {
+				allErrs = append(allErrs, field.Invalid(vpnNetwork.GetFieldPath(), vpnNetwork.GetCIDR(), "VPN network must have a /24 prefix for IPv4 seeds"))
+			}
+			if primaryIPFamily == core.IPFamilyIPv6 && prefix != 120 {
+				allErrs = append(allErrs, field.Invalid(vpnNetwork.GetFieldPath(), vpnNetwork.GetCIDR(), "VPN network must have a /120 prefix for IPv6 seeds"))
+			}
```

Comment:
You're right, we could have implemented it this way. However, in https://github.com/gardener/vpn2/pull/78 kept it simple and ensures that `VPN_NETWORK` is set to a `/24` or `/120` net: https://github.com/gardener/vpn2/blob/08a3f75ad01e9e249d7319eeb984b5b2125aa05d/seed-server/network-connection.sh#L50-L53

We can still lift this restriction if there is a need. But it will require some more bash magic in vpn2 😄